mmda/types/document.py

```diff
@@ -107,20 +107,24 @@ def to_json(self, fields: Optional[List[str]] = None, with_images=False) -> Dict
                 field2: [...]
             }
         """
+        doc_dict = {Symbols: self.symbols}
+
+        # figure out which fields to serialize
         fields = self.fields if fields is None else fields              # use all fields unless overridden
+        fields = [field for field in fields if field != Symbols]
         if not with_images:
             fields = [field for field in fields if field != Images]     # remove images from considered fields
-        return {
-            field: [doc_span_group.to_json() for doc_span_group in getattr(self, field)]
-            for field in fields
-        }
+
+        # add to doc dict
+        for field in fields:
+            doc_dict[field] = [doc_span_group.to_json() for doc_span_group in getattr(self, field)]
+
+        return doc_dict
 
     @classmethod
     def from_json(cls, doc_dict: Dict) -> "Document":
-        fields = doc_dict.keys()        # TODO[kylel]: this modifies the referenced dict, not copy
```

Comment:
Do we want to add `doc_dict = doc_dict.copy()` to avoid changing the input?

Reply:
i changed `self.fields` to just contain added fields.  we shouldnt consider `Symbols` or `Images` fields; they're more just core data representation

Reply:
this also makes the `to_json` `from_json` methods to be easier wrt logic -- we have explicit calls to Symbols Images wrt doc_dict manipulation; and loop through all additional fields otherwise